ui/pages/confirmation/templates/add-ethereum-chain.js

```diff
@@ -331,7 +347,35 @@ function getValues(pendingApproval, t, actions, history) {
     ],
     cancelText: t('cancel'),
     submitText: t('approveButtonText'),
+    loadingText: t('addingCustomNetwork'),
     onSubmit: async () => {
+      let endpointChainId;
+      try {
+        endpointChainId = await jsonRpcRequest(customRpcUrl, 'eth_chainId');
+      } catch (_err) {
+        // eslint-disable-next-line no-console
+        console.error(
+          `Request for method 'eth_chainId on ${customRpcUrl} failed`,
+        );
+        return actions.rejectPendingApproval(
+          pendingApproval.id,
+          ethErrors.rpc.resourceUnavailable().serialize(),
+        );
+      }
+
+      if (pendingApproval.requestData.chainId !== endpointChainId) {
+        console.error(
+          `Chain ID returned by RPC URL ${customRpcUrl} does not match ${endpointChainId}`,
+        );
+        // actions.rejectPendingApproval(
+        //   pendingApproval.id,
+        //   ethErrors.rpc.invalidInput().serialize(),
+        // );
```

Comment:
Do we still want to be rejecting the pending approval? Or do we no longer need this now that the user is forced in to clicking cancel themselves